src/helper.cc

```diff
@@ -1045,12 +1045,14 @@ helperHandleRead(const Comm::ConnectionPointer &conn, char *, size_t len, Comm::
 
     if (!srv->stats.pending && !srv->stats.timedout) {
         /* someone spoke without being spoken to */
-        debugs(84, DBG_IMPORTANT, "ERROR: helperHandleRead: unexpected read from " <<
+        debugs(84, DBG_IMPORTANT, "ERROR: Killing helper process after an unexpected read from " <<
                hlp->id_name << " #" << srv->index << ", " << (int)len <<
                " bytes '" << srv->rbuf << "'");
```

Comment:
This and the matching `debugs()` on line 1168  in `helperStatefulHandleRead()` probably should be converted to use the new `cache_log_message` feature in case the admin has no control over a third-party helper.

Also their `srv->rbuf` display should be using `Raw()` to a) prevent log injection issues, and b) hide the raw I/O bytes down at level `DBG_DATA`

Reply:
Can we do this separately in a follow-up PR, like a recent b8aba8 for from-helper annotations?

> Also their srv->rbuf display should be using Raw()

IMO, this also looks as a separate fix deserving a another PR.